docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,247 @@
+---
+layout: global
+title: TRANSFORM
+displayTitle: TRANSFORM
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+The `TRANSFORM` clause is used to specify a Hive-style transform (`SELECT TRANSFORM`/`MAP`/`REDUCE`)
+query specification to transform the input by running a specified script. Users can
+plug in their own custom mappers or reducers in the data stream by using features natively supported
+in the Spark SQL. In order to run a custom mapper script `map_script` or a custom
+reducer script `reduce_script` the user can issue the command which uses the `TRANSFORM`
+clause to embed the mapper or the reducer scripts.
+
+### Syntax
+
+```sql
+SELECT { TRANSFORM ( named_expression [ , ... ] ) | MAP named_expression [ , ... ] | REDUCE named_expression [ , ... ] }
```

Comment:
is it simply `... { TRANSFORM | MAP | REDUCE } ( named_expression [ , ... ] )`?

Reply:
> is it simply `... { TRANSFORM | MAP | REDUCE } ( named_expression [ , ... ] )`?

In spark, there's no difference when execution

Reply:
> is it simply `... { TRANSFORM | MAP | REDUCE } ( named_expression [ , ... ] )`?

No, since
```
transform (a, b, c)
map a, b, c
reduce a, b, c
```

Reply:
ah, then
```
... { TRANSFORM ( named_expression [ , ... ] ) | { MAP | REDUCE } named_expression [ , ... ] }
```

Reply:
shall we forbid MAP REDUCE in Spark? The behavior doesn't match Hive and also not very reasonable.

Reply:
> shall we forbid MAP REDUCE in Spark? The behavior doesn't match Hive and also not very reasonable.

The results of their execution should be the same. If this syntax is forbidden, it will cause compatibility problems. Let me check more about this. Then come to the conclusion

Reply:
In Hive, `MAP` means `SELECT TRANSFORM`, I think our syntax is incorrect.

Reply:
See https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Transform

Reply:
At least we should deprecate it and not document it.

Reply:
> In Hive, `MAP` means `SELECT TRANSFORM`, I think our syntax is incorrect.

My fault, here's syntax I write is wrong, it should be 
```
{ SELECT TRANSFORM ( named_expression [ , ... ] ) | { MAP | REDUCE } named_expression [ , ... ] }
  
```